### PR TITLE
MRIID-5 Added covid data to chart

### PR DESCRIPTION
**Pull request by fernandodonaire**
Screen Shot 2020-12-04 at 2 14 15 PM

Can be tested by:

1) Pulling down this branch.
2) Running the app locally.
3) Selecting COVID 19 in the Sidebar. You should see the COVID case count data in the chart.

### CODE REVIEW

File: src/utils/__tests__/chartDataHelpers.test.js [hunk 1]
```diff
@@ -100,8 +103,47 @@ describe("Tests for the chart data helper functions", () => {
...
-    expect(prepareCovidDataForCharts()).toEqual([
+  test("prepareCovidDataForCharts returns all country data in expected format", () => {
+    const covidAllCountriesFilters = {
+      ...reduxInitialState.filters,
+      outbreak: "COVID 19",
+      dateRange: covidInitialDateRange,
+    };
+    expect(
+      prepareCovidDataForCharts(
+        null,
```
Comment (SpencerMarks-IDHA) [inline src/utils/__tests__/chartDataHelpers.test.js:114]: we should avoid designing APIs that accept null as valid. etc.

Comment (SpencerMarks-IDHA) [reply src/utils/__tests__/chartDataHelpers.test.js:114]: or maybe this is a negative test case?

File: src/utils/__tests__/chartDataHelpers.test.js [hunk 2]
```diff
@@ -100,8 +103,47 @@ describe("Tests for the chart data helper functions", () => {
...
+        testCovidDataCombined.data,
+        covidAllCountriesFilters
+      )
```
Comment (SpencerMarks-IDHA) [inline src/utils/__tests__/chartDataHelpers.test.js:117]: in any case, great to see test cases.

File: src/utils/chartDataHelpers.js [hunk 3]
```diff
@@ -126,9 +130,43 @@ export const prepareEbolaDataForCharts = (
...
+  // Add column headers to chartData array.
   chartData.push(getChartColumns("COVID-19", false));
+  if (getAllCountriesCovidData) {
+    // Get an array of keys from the covidDataCombined.cases object where the dates are 7 days apart.
+    const weekDateKeys = getWeeklyDateObjectKeys(covidDataCombined.cases);
+    // Loop through all the 'weekDateKeys'.
+    weekDateKeys.forEach((dateKey) => {
+      // If the 'dateKey' is within the dates in the filters, push the data row to the chartData array.
+      if (isDateWithinFiltersDateRange(dateKey, filters.dateRange)) {
+        chartData.push([new Date(dateKey), covidDataCombined.cases[dateKey]]);
```
Comment (SpencerMarks-IDHA) [inline src/utils/chartDataHelpers.js:152]: the complexity here makes the code hard to follow. The comments are great boon, but I wonder if you could break this up into methods or otherwise refactored?

**Review by SpencerMarks-IDHA — CHANGES_REQUESTED** [review]
One change request,  but only if feasible.

File: src/utils/chartDataHelpers.js (hunk 3 above)
Comment (fernandodonaire) [reply src/utils/chartDataHelpers.js:152]: Yeah good call. I can break up the `prepareCovidDataForCharts()` function into 2 separate functions. One to get 'all countries' data and another to get the country specific data.

Comment (fernandodonaire) [reply src/utils/chartDataHelpers.js]: Hey @SpencerMarks-IDHA , I refactored the `prepareCovidDataForCharts()` function. Can you please review this again?

**Review by SpencerMarks-IDHA — APPROVED** [review]
